src/proc/y411-converter.cpp

```diff
@@ -0,0 +1,311 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+#include "y411-converter.h"
+
+#ifdef RS2_USE_CUDA
+#include "cuda/cuda-conversion.cuh"
+#endif
+#ifdef __SSSE3__
+#include <tmmintrin.h> // For SSSE3 intrinsics
+#endif
+
+namespace librealsense 
+{
+    void convert_yuv_to_rgb(const byte yuv[3], byte * rgb)
+    {
+        int32_t c = yuv[0] - 16;
+        int32_t d = yuv[1] - 128;
+        int32_t e = yuv[2] - 128;
+
+        int32_t t;
+#define clamp( x ) ( ( t = ( x ) ) > 255 ? 255 : t < 0 ? 0 : t )
+        rgb[0] = clamp((298 * c + 409 * e + 128) >> 8);
+        rgb[1] = clamp((298 * c - 100 * d - 208 * e + 128) >> 8);
+        rgb[2] = clamp((298 * c + 516 * d + 128) >> 8);
+#undef clamp
+    }
+
+    // The bytes alignment on y411:
+    // Y is luminance and U,V are chrome
+    // each U,V are duplicated for 4 pixels
+    //
+    // [u1 y1 y2 v1 y3 y4]    [u2 y5 y6 v2 y7 y8]   [u3 y9 y10 v3 y11 y12]
+    //
+    // Before converting to RGB we unpack the y411 to yuv and then convert it to RBG
+    // After the transformation to yuv the bytes alignment looks like this:
+    //
+    // [y1 u1 v1][y2 u1 v1]     [y5 u2 v2][y6 u2 v2]    [y9 u3 v3][y10 u3 v3]
+    // [y3 u1 v1][y4 u1 v1]     [y7 u2 v2][y7 u2 v2]    [y11 u3 v3][y12 u3 v3]
+    //
+    //https://www.fourcc.org/pixel-format/yuv-y411/
+
+    void inline unpack_y411_sse(byte * const dest, const byte * s, int w, int h, int actual_size)
```

Comment:
Put the whole function inside the SSE #ifdef

Reply:
Why inline?

Reply:
done